### PR TITLE
LPS-102867 Select Options were shifted for every Repeatable Field

### PR DESCRIPTION
**Pull request by leoadb**

### CODE REVIEW

**Comment by leoadb** [comment]
@natocesarrego Can you please help me out with this? Please see the comments in https://github.com/pavel-savinov/liferay-portal/pull/447#discussion_r333565448. Thanks!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js [hunk 1]
```diff
@@ -498,6 +498,12 @@ AUI.add(
 					if (
 						currentTarget.hasClass('lfr-ddm-repeatable-add-button')
 					) {
+						var field = instance.getFieldDefinition();
+
+						if (field.type === 'select' && field.options[0].value === '') {
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js]: Hi @matthewchan1,

Why did you need to and this extra check `field.options[0].value === ''`?

**Comment by natocesarrego** [comment]
Hi @matthewchan1,

You fix makes sense to me and I just have a comment regarding it, can you please check my comment?

Besides, a conflict was reported here, can you please check it too?

Lastly, sorry for the delay to answer you, but I was out of office a few days and my current workload is considerably high.

Thanks.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js (hunk 1 above)
Comment (matthewchan1) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js]: Hi @natocesarrego, the extra check is insurance in case the call to repeat() is triggered before Select Field's getOptions(), which creates the placeholder with ''. It's a very unlikely edge case since rendering the page uses getOptions, so if you'd like, we can remove that added check.

Comment (natocesarrego) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-web/src/main/resources/META-INF/resources/js/ddm_form.js]: If you don't find a valid use case for it, @matthewchan1, then I believe we should remove it. Otherwise, keep it. In addition, can you please rebase and resend due to conflicts? Thanks.
